vapi/vcenter/consumptiondomains/associations/associations.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright (c) 2024-2024 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package associations
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+
+	"github.com/vmware/govmomi/vapi/rest"
+)
+
+const (
+	basePath         = "/api/vcenter/consumption-domains/zones/cluster"
+	associationsPath = basePath + "/%s/associations"
+)
+
+// Manager extends rest.Client, adding vSphere Zone association related methods.
+type Manager struct {
+	*rest.Client
+}
+
+// NewManager creates a new Manager instance with the given client.
+func NewManager(client *rest.Client) *Manager {
+	return &Manager{
+		Client: client,
+	}
+}
+
+// AddAssociations
+// https://developer.broadcom.com/xapis/vsphere-automation-api/latest/vcenter/api/vcenter/consumption-domains/zones/cluster/zone/associations__action=add/post
+func (c *Manager) AddAssociations(zone string, clusters []string) error {
+	path := c.Resource(fmt.Sprintf(associationsPath, zone)).WithParam("action", "add")
+	req := path.Request(http.MethodPost, clusters)
+	var res map[string]interface{}
+	if err := c.Do(context.Background(), req, &res); err != nil {
+		return err
+	}
+
+	// This endpoint does not always return and error upon failure.
+	// In such cases we need to parse the response to figure out the actual status.
+
+	success := res["success"].(bool)
```

Comment:
This would panic if `success` isn't in the map, can prevent that:
```go
	success, ok := res["success"].(bool)
	if !(ok && success) {
		return errors.New(...)
	}
```

Reply:
thanks, fixed